src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -2529,6 +2531,31 @@ void QueryAnalyzer::expandOrderByAll(QueryNode & query_tree_node_typed, const Se
     query_tree_node_typed.setIsOrderByAll(false);
 }
 
+void QueryAnalyzer::expandLimitByAll(QueryNode & query_tree_node_typed, const Settings & settings)
+{
+    if (!settings.enable_limit_by_all || !query_tree_node_typed.isLimitByAll())
+        return;
+
+    auto & projection_nodes = query_tree_node_typed.getProjection().getNodes();
+    auto list_node = std::make_shared<ListNode>();
+    list_node->getNodes().reserve(projection_nodes.size());
+
+    for (auto & node : projection_nodes)
+    {
+        auto resolved_expression_it = resolved_expressions.find(node);
```

Comment:
Actually, this seems to be wrong. Projection list may be unresolved at this moment in case of `group_by_use_nulls`